shared/src/main/java/com/faforever/neroxis/mask/Mask.java

```diff
@@ -811,19 +697,50 @@ protected Map<Integer, Integer> getShiftedCoordinateMap(int offset, boolean cent
                         .collect(Collectors.toMap(i -> i, i -> getShiftedValue(i, trueOffset, toSize, wrapEdges)));
     }
 
-    protected void loopWithSymmetry(SymmetryType symmetryType, BiIntConsumer maskAction) {
+    protected void loop(BiIntConsumer maskAction) {
         assertNotPipelined();
-        int minX = getMinXBound(symmetryType);
-        int maxX = getMaxXBound(symmetryType);
-        for (int x = minX; x < maxX; x++) {
-            int minY = getMinYBound(x, symmetryType);
-            int maxY = getMaxYBound(x, symmetryType);
+        int size = getSize();
+        for (int x = 0; x < size; x++) {
+            for (int y = 0; y < size; y++) {
+                maskAction.accept(x, y);
+            }
+        }
+    }
+
+    protected void loopInSymmetryRegion(SymmetryType symmetryType, BiIntConsumer maskAction) {
+        assertNotPipelined();
+        Symmetry symmetry = symmetrySettings.getSymmetry(symmetryType);
+        int size = getSize();
+        int maxX = SymmetryUtil.getMaxXBound(symmetry, size);
+        IntUnaryOperator minYBoundFunction = SymmetryUtil.getMinYBoundFunction(symmetry, size);
+        IntUnaryOperator maxYBoundFunction = SymmetryUtil.getMaxYBoundFunction(symmetry, size);
+        for (int x = 0; x < maxX; x++) {
+            int minY = minYBoundFunction.applyAsInt(x);
+            int maxY = maxYBoundFunction.applyAsInt(x);
```

Comment:
So I've been playing a bit in this branch, and testing it out.. It's looking pretty good...
I needed to do this to fix some missing pxiels.

`int maxY = StrictMath.min(maxYBoundFunction.applyAsInt(x) +1, size);`